Lesson33/props-drilling-app/Gokhan/src/components/UserDetailsForm/index.jsx

```diff
@@ -0,0 +1,22 @@
+import './styles.css';
+
+export const UserDetailsForm = ({ username, onUsernameChange }) => {
+
+  const handleChange = (e) => {
+    onUsernameChange(e.target.value);
```

Comment:
You should not update the actual username value (that comes from App component) here as the change may be temporary (unwanted).
Add local state to this component:
`const [tempUsername, setTempUsername] = useState(username);`

then update the state in the handleChange function:
`setTempUsername(e.target.value)` instead of `onUsernameChange(e.target.value);`

Only when the form is submitted, call onUsernameChange function. It's better not to add onClick event listener to the button but add onSubmit event listener to the form element instead.